### PR TITLE
add method is_simple to permutations

### PR DESCRIPTION
**Pull request by fchapoton**
### :books: Description

Add one method to permutations, checking if it is simple (not sending any proper sub-interval to a sub-interval)

Also fixing a few pep8 warnings in the modified file on the way.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Review by kevindilks — COMMENTED** [review]
I'd like to see a little more in the description/examples so that a user can more readily verify that they understand the definition correctly without having to go to the OEIS link.

Description could mention that [6,1,3,5,2,4] is not simple because it maps the interval [3,4,5,6] to [2,3,4,5], whereas [2,6,3,5,1,4] is simple. And then have those both be tests/examples (along with the cardinality check).

**Comment by fchapoton** [comment]
thanks for the review. I have added your suggestions.

**Comment by videlec** [comment]
Isn't there a linear algorithm? At least the following is sub-quadratic
```python
def is_simple(self):
    r"""
    EXAMPLES::

            sage: [sum(is_simple(pi) for pi in Permutations(n))  for n in range(8)]
            [1, 1, 2, 0, 2, 6, 46, 338]
    """
    n = len(self)
    if n <= 2:
        return True

    # testing intervals starting at position 0
    start = 0
    left = right = self._list[0]
    end = 1
    while end < n - 1:
        elt = self._list[end]
        if elt < left:
            left = elt
        elif elt > right:
            right = elt
        if right - left == end - start:
            return False
        elif right - left == n - 1:
            break
        end += 1

    # testing intervals starting at later positions
    for start in range(1, n - 1):
        left = right = self._list[start]
        end = start + 1
        while end < n:
            elt = self._list[end]
            if elt < left:
                left = elt
            elif elt > right:
                right = elt
            if right - left == end - start:
                return False
            elif right - left > n - 1 - start:
                break
            end += 1
            
    return True
```

**Comment by fchapoton** [comment]
oui, c'est déjà mieux. Il faudrait chercher dans la littérature si quelqu'un a un meilleur algo. On peut aussi mettre le code en cython.

File: src/sage/combinat/permutation.py [hunk 1]
```diff
@@ -3094,6 +3094,76 @@ def is_derangement(self) -> bool:
...
+            sage: g.is_simple()
+            False
+
+            sage: g = Permutation([2,6,3,5,1,4])
+            sage: g.is_simple()
+            True
+
+        TESTS::
+
+            sage: [len([pi for pi in Permutations(n) if pi.is_simple()])
```
Comment (videlec) [inline src/sage/combinat/permutation.py:3124]: Why is this only a TEST? It is relevant for the user (not the most efficient way to get the list of simple permutations though).

Comment (fchapoton) [reply src/sage/combinat/permutation.py:3124]: Comme tu veux. Mais je ne vois pas ce que ca apporte. On a le lien OEIS.

Comment (videlec) [reply src/sage/combinat/permutation.py:3124]: Précisément, c'est un lien, il faut cliquer dessus pour avoir l'info.

Comment (fchapoton) [reply src/sage/combinat/permutation.py:3124]: Voilà. On cythonise ou pas ? J'ai la flemme, et d'autres occupations plus pressantes.

Comment (videlec) [reply src/sage/combinat/permutation.py:3124]: Je vois pas trop l'intérêt à ce stade.

**Review by videlec — APPROVED** [review]

**Comment by fchapoton** [comment]
merci Vincent.

**Pull request merged**